src/Fixer/Phpdoc/PhpdocTypesFixer.php

```diff
@@ -139,18 +139,23 @@ protected function configurePostNormalisation(): void
 
     protected function normalize(string $type): string
     {
-        $typeLower = strtolower($type);
-        if (isset($this->typesSetToFix[$typeLower])) {
-            $type = $typeLower;
-        }
-
-        // normalize shape/callable/generic identifiers too
-        // TODO parse them as inner types and this will be not needed then
-        return Preg::replaceCallback(
-            '/^(\??\s*)([^()[\]{}<>\'"]+)(?<!\s)(\s*[\s()[\]{}<>])/',
-            fn ($matches) => $matches[1].$this->normalize($matches[2]).$matches[3],
-            $type
-        );
+        $typeExpression = new TypeExpression($type, null, []);
+
+        $newTypeExpression = $typeExpression->mapTypes(function (TypeExpression $type) {
+            if ($type->isUnionType()) {
```

Comment:
Let's finish #8210 first and not introduce any more confusing usages of `isUnionType()`. But if you prefer to finish this one first then fine, but this condition has to be fixed in the other PR.

Reply:
marking as resolved as there are no conflicts - `isUnionType` was kept